src/utilities/modules/vaccines.js

```diff
@@ -413,33 +415,40 @@ export const extractDataForBreachModal = ({
 };
 
 /**
- * Returns chart props for fridge chart
+ * Returns chart props for fridge chart.
+ *
+ * Collects the last 30 days of sensor logs for a given fridge.
+ * aggregates these logs into periods. Will use a number of periods
+ * in [ 0, 30 ] - attempting to use 9 hour periods. If there are less
+ * sensorLogs than 9 hour periods, we just use the sensorlogs length.
+ *
+ * @param {Realm}        database App wide database
+ * @param {Realm.Fridge} fridge   Provided fridge object to find chart data for.
  */
 export const extractDataForFridgeChart = ({ database, fridge }) => {
+  // Fetch the last 30 days of sensorlogs
   const sensorLogs = fridge.getSensorLogs(database, FRIDGE_CHART_LOOKBACK_MS);
-
-  const chartRangeMilliseconds = sensorLogs.max('timestamp') - sensorLogs.min('timestamp');
-
-  let numberOfDataPoints = 0;
-  // Full aggregation is 12 hour periods
-  if (chartRangeMilliseconds < 12 * 60 * 60 * 1000) {
-    // Pre aggregation is in 20 minute periods
-    numberOfDataPoints = Math.floor(chartRangeMilliseconds / (20 * 60 * 1000));
-  } else {
-    numberOfDataPoints = Math.floor(chartRangeMilliseconds / (12 * 60 * 60 * 1000));
-  }
-
-  numberOfDataPoints = numberOfDataPoints > 30 ? 30 : numberOfDataPoints;
-
+  // Find the duration of of all sesorlogs and the current date. Leaves space
+  // in the chart indicating when the last sync cycle was.
+  const chartRangeMilliseconds = new Date().getTime() - sensorLogs.min('timestamp');
+  // Find the number of nine hour intervals during the duration of sensorlogs.
+  const nineHourIntervals = Math.floor(chartRangeMilliseconds / FRIDGE_CHART_DATA_POINT_INTERVAL);
+  // Use the maximum of either nine hour intervals, or the length of sensorlogs.
+  // If there are not enough sensorlogs to have a datapoint for each 9 hour interval,
+  // just use a datapoint for each sensorlog.
+  const chartIntervals = Math.max(nineHourIntervals, sensorLogs.length);
```

Comment:
Good point. I don't know what you mean by "data point count" though? Is that `MAX_FRIDGE_CHART_DATAPOINTS`?

Alternative could be to filter out all `breachAggregates` from `sensorLogs`

Reply:
@joshxg filtering breachAggregates would remove current ongoing breach from the graph. I was meaning, if sensorLog data is < FRIDGE_CHART_DATA_POINT_INTERVAL then use 'sensorLogs.length', how does that sound ?

Reply:
Didn't read properly, this was meant to be `Math.min` oops, but there is still a problem, when either there is such little data i.e. a couple of hours, there will be one data point. 

@andreievg so we don't use 9 hour intervals at all? Just use `const numberOfDataPoints = Math.min(sensorLogs.length, MAX_FRIDGE_CHART_DATA_POINTS)`? This is how `extractDataForBreachModal` works, I believe. I think this only works for that method because of the interval between logs is consistent.

Consider: 23 hours of data - 16 hours is aggregated, leaving ~2 full aggregation logs for the first 16 hours, and ~21 logs for the most recent ~7 hours (assuming no breaches). When `aggregateLogs` is run: we use 23 data points and will have gaps throughout the first 20 hours of logs (this is why my preference is to keep `preAggregate` logs for displaying charts as the consistent interval between logs  makes things a lot simpler).

Basically, if the number of `fullAggregationLogs < preAggregateLogs + breachAggregateLogs`, we get gaps (using length - consider having 10 `fullAggregateLogs` and 23 preAggregate logs, we use 30 data points, we would have gaps). This should be the case for the first ~8 days.

Unless, we use 9 hour intervals for this condition? However this does mean a limited graph for the first little while...


```
if (fullAggregationLogs.length < 0)
   dataPoints = 20minuteIntervals
else if(fullAggregationLogs.length < preAggregateLogs.length + breachAggregateLogs.length and also less than 30)
   dataPoints = 9 hour intervals
else
   dataPoints = 30
```

Hopefully this makes some sense..

Reply:
@joshxg 
That does kind of make sense. Can we simplify it further by:
* When doing full aggregation, don't delete preAggregates that are X days old ?
* When doing full aggregation start at preAggregates that are newer then last full aggregate
* When sensorLogs (with aggregation != '') are more then X days, use 9 hour intervals, otherwise 30min Intervals ?